### PR TITLE
[docs] - Re-organize Concepts category page

### PR DESCRIPTION
**Pull request by erinkcochran87**
## Summary & Motivation

This PR updates the Concepts category page to match the recently re-vamped sidenav. Specifically, it adds a new **Advanced** section and moves several concept subsections under it.

I noticed I missed this while working on #16156.

## How I Tested These Changes

👀 , local

### CODE REVIEW

**Comment by erinkcochran87** [comment]
@jamiedemaria Could I get your 👀 on this?

File: docs/content/concepts.mdx [hunk 1]
```diff
@@ -8,9 +8,9 @@ Learn about Dagster's core concepts and how to use them in your data platform.
 
 ---
 
-## Software-defined assets
+## Software-defined Assets
 
-An asset is an object in persistent storage, such as a table, file, or persisted machine learning model. A software-defined asset is a Dagster object that couples an asset to the function and upstream assets used to produce its contents.
+An asset is an object in persistent storage, such as a table, file, or persisted machine learning model. A Software-defined Asset is a Dagster object that couples an asset to the function and upstream assets used to produce its contents.
 
 <ArticleList>
```
Comment (jamiedemaria) [inline docs/content/concepts.mdx:15]: Looks like the graph backed asset page never got added to this list, could you add it as part of this PR?

File: docs/content/concepts.mdx [hunk 2]
```diff
@@ -37,66 +37,6 @@ An asset is an object in persistent storage, such as a table, file, or persisted
...
-  ></ArticleListItem>
-  <ArticleListItem
-    title="Metadata and tags"
-    href="/concepts/ops-jobs-graphs/metadata-tags"
-  ></ArticleListItem>
-</ArticleList>
-
----
-
 ## Schedules and sensors
```
Comment (jamiedemaria) [inline docs/content/concepts.mdx:40]: and looks like AsestSensors never got added to this list

File: docs/content/concepts.mdx [hunk 3]
```diff
@@ -131,86 +71,134 @@ A software-defined asset or job can represent a collection of _partitions_ that
 
 ---
 
-## I/O management
+## Resources & configuration
+
+Resources enable you to separate logic from external dependencies, making developing and testing possible in multiple environments. [Learn more about resources](/concepts/resources).
```
Comment (jamiedemaria) [inline docs/content/concepts.mdx:76]: how does this look if the Resources page is a bullet like all of the other sub sections? It was a tiny bit strange to see the link for the resources page as part of a sentence when all of the other links are bullets

Comment (jamiedemaria) [reply docs/content/concepts.mdx:76]: or maybe just having the "Learn more about resources" be on a new line like the Dagster UI section

**Review by jamiedemaria — COMMENTED** [review]
cool, looks good! I noticed a couple pages that never made it into this landing page, but other than that this looks very nice!!

**Comment by erinkcochran87** [comment]
@jamiedemaria Back to you! I also updated the Assets sidenav to match the category page, which includes the re-ordering we talked about yesterday.

**Review by jamiedemaria — APPROVED** [review]
yay!

**Pull request merged**